North-Seattle-College/ad440-winter2022-thursday-repo
# Test the React app for cross-site scripting (issue 132)

### Closes issue #132 

### Overview
- This pull request adds `testCrossSiteScripting.py`
- This file inspects HTTP security headers and runs POST requests with common cross site scripting (XSS) tests.  The following arguments can be provided at runtime:
   - --url: target URL to check for vulnerabilities [optional]
- If not URL argument is provided, a default URL will be used: [https://9u4xt4nqr1.execute-api.us-west-2.amazonaws.com/default/test'](url)

### Pre-requisites
- Python requests library: `python -m pip install requests`

### To Test
1. Open the branch: `132/nwc/feature/xss-testing`
2. Run the script with an optional argument to confirm functionality: `python testCrossSiteScripting.py --url https://9u4xt4nqr1.execute-api.us-west-2.amazonaws.com/default/test`
4. Script should run and output the results of testing to the terminal:
![testCrossSiteScripting_py_—_ad440-winter2022-thursday-repo](https://user-images.githubusercontent.com/45498644/158568275-a77ff7f1-2268-4c33-b0b0-30b6c2b24ce3.png)

### Time Spent
| Activity | Estimate | Actual Time Spent |
| ------ | ------ | ------ |
| Research cross-site scripting attacks and testing | 3 hours | 4 hours |
| Write testing script | 3 hours | 4 hours |
| Test script   | 2 hours | 1 hour |
| Write documentation | 30 mins | 30 mins |
| Pull request and wrap-up | 30 mins | 30 mins |
